osf/utils/workflows.py

```diff
@@ -94,6 +114,57 @@ def values(cls):
     }
 ]
 
+REGISTRATION_TRANSITIONS = [
+    {
+        'trigger': DefaultTriggers.SUBMIT.value,
+        'source': [DefaultStates.INITIAL.value],
+        'dest': DefaultStates.PENDING.value,
+        'after': ['save_action', 'update_last_transitioned', 'submit_draft_registration', 'notify_submit'],
+    },
+    {
+        'trigger': DefaultTriggers.ACCEPT.value,
+        'source': [DefaultStates.PENDING.value, DefaultStates.REJECTED.value],
+        'dest': DefaultStates.ACCEPTED.value,
+        'after': ['save_action', 'update_last_transitioned', 'accept_draft_registration', 'notify_accept_reject'],
+    },
+    {
+        'trigger': DefaultTriggers.REJECT.value,
+        'source': [DefaultStates.PENDING.value, DefaultStates.ACCEPTED.value],
```

Comment:
I think rejecting from an accepted state should be a force-withdrawal, per https://openscience.atlassian.net/browse/REQS-127, rather than going into a rejected state. We didn't have this transition on the state diagram.